unified_planning/model/ma_problem.py

```diff
@@ -0,0 +1,250 @@
+                                setattr(i.fluent(), '_name',
+                                        str(getattr(i.fluent(), '_name')) + "_" + str(self.get_agents().index(ag)))
+                                #print("quiiiii", i.fluent()._name)
+                                self.fluents_name.append(i.fluent()._name)
+
+
+
+                    '''if i._content.payload:
+                        print("quiiiii", i._content.payload)
+                        setattr(i.fluent(), '_name',
+                                str(getattr(i.fluent(), '_name')) + "_" + str(self.get_agents().index(ag)))'''
+                    '''if i._content.args != None:
+                        for flu in i._content.args:
+                            if flu.is_fluent_exp():
+                                setattr(flu.fluent(), '_name', str(getattr(flu.fluent(), '_name')) + "_" + str(self.get_agents().index(ag)))
+                    if i.is_fluent_exp():
+                        setattr(i._content.payload, '_name', str(getattr(i._content.payload, '_name')) + "_" + str(self.get_agents().index(ag)))'''
+                self.add_action(act)
+            for flu, value in ag.get_initial_values().items():
+                flu = copy.deepcopy(flu)        #hash commentato in fnode
+                value = copy.deepcopy(value)
+                setattr(flu.fluent(), '_name', str(getattr(flu.fluent(), '_name')) + "_" + str(self.get_agents().index(ag)))
+                self.set_initial_value(flu, value)
+            for goal in ag.get_goals():
+                goal = copy.deepcopy(goal)
+                if goal.is_fluent_exp():
+                    setattr(goal.fluent(), '_name', str(getattr(goal.fluent(), '_name')) + "_" + str(self.get_agents().index(ag)))
+                self.add_goal(goal)
+            '''for obj in ag.get_all_objects():
+                obj = copy.deepcopy(obj)
+                print("OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOObj", obj)
+                setattr(obj, '_name', str(getattr(obj, '_name')) + "_" + str(self.get_agents().index(ag)))
+                self.add_object(obj)'''
+        return self
+
+
+
+    def solve_OneshotPlanner(self):
+        with OneshotPlanner(name='pyperplan') as planner:
+            solve_plan = planner.solve(self)
+            print("Pyperplan returned: %s" % solve_plan)
+        return
+
+    def extract_plans(self, plan_problem):
+        for ag in self.get_agents():
+            for act in plan_problem._actions:
+                act = copy.deepcopy(act)
+                setattr(act._action, 'name', str(getattr(act._action, 'name')) + "_" + str(self.get_agents().index(ag)))
+                #for par in act._params: #l1, l2 , l1_0. l2_1 ...
+                #    setattr(par._content.payload, '_name', str(getattr(par._content.payload, '_name')) + "_" + str(self.get_agents().index(ag)))
+                self.plan.append(act)
+        return self.plan
+
+    def pddl_writer(self):
```

Comment:
It would be better to have a dedicated writer class in the `unified_planning.io` for this